plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/exporter/SpanExporterFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.exporter;
+
+import io.opentelemetry.exporter.logging.LoggingSpanExporter;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.lang.reflect.Method;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Factory class to create the {@link SpanExporter} instance.
+ */
+public class SpanExporterFactory {
+
+    private static final Logger logger = LogManager.getLogger(SpanExporterFactory.class);
+
+    /**
+     * Span Exporter type setting.
+     */
+    @SuppressWarnings("unchecked")
+    public static final Setting<Class<SpanExporter>> OTEL_TRACER_SPAN_EXPORTER_CLASS_SETTING = new Setting<>(
```

Comment:
We may consider consolidating settings in on place (fe `OtelTelemetrySettings`) since they are now become scattered across different places (we have examples in repository plugins).